#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_fix_pkg_rebuild.m.dpatch by Thomas Weber <thomas.weber.mail@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix -rebuild issue with pkg.m, patch taken from CVS, version 1.47
## DP: http://www.nabble.com/Small-bug-in-%22pkg-rebuild%22-tf3807246.html

@DPATCH@
diff -urNad octave-2.9.12~/scripts/pkg/pkg.m octave-2.9.12/scripts/pkg/pkg.m
--- octave-2.9.12~/scripts/pkg/pkg.m	2007-05-22 19:25:29.000000000 +0000
+++ octave-2.9.12/scripts/pkg/pkg.m	2007-05-24 18:24:31.000000000 +0000
@@ -291,7 +291,7 @@
 
     case "rebuild"
       if (global_install)
-	global_packages = rebuild (prefix, global_list, auto, verbose);
+	global_packages = rebuild (prefix, global_list, files,  auto, verbose);
 	save (global_list, "global_packages");
 	local_packages = global_packages;
       else
@@ -315,9 +315,12 @@
   else
     old_descriptions = installed_packages (list, list);
     wd = pwd ();
-    cd (prefix);
-    dirlist = glob (cellfun(@(x) [x, '-*'], files, 'UniformOutput', 0))
-    cd (wd);
+    unwind_protect
+      cd (prefix);
+      dirlist = glob (cellfun(@(x) [x, '-*'], files, 'UniformOutput', 0))
+    unwind_protect_cleanup
+      cd (wd);
+    end_unwind_protect
   endif
   descriptions = {};
   for k = 1:length (dirlist)
@@ -351,10 +354,10 @@
     endif
   endfor
 
-  if (isempty (files))
+  if (! isempty (files))
     ## We are rebuilding for a particular package(s) so we should take
     ## care to keep the other untouched packages in the descriptions
-    descriptions = {desriptions{:}, old_desriptions{:}};
+    descriptions = {descriptions{:}, old_descriptions{:}};
 
     dup = [];
     for i = 1:length (descriptions)
